Moving dnssec keygen to ansible task

gekios opened the pull request:
Temporarily moving dnssec key generation from deploy-teuthology script to
teuthology_nameserver ansible role due to current encryption algorithm
being depricated in Tumbleweed

Signed-off-by: Georgios Kyratsas <gkyratsas@suse.com>

Commit:
Moving dnssec keygen to ansible task

Temporarily moving dnssec key generation from deploy-teuthology script to
teuthology_nameserver ansible role due to current encryption algorithm
being depricated in Tumbleweed

Signed-off-by: Georgios Kyratsas <gkyratsas@suse.com>

kshtsk commented on deploy-teuthology:
```diff
@@ -173,19 +173,17 @@ fi
-PUBKEY_NAME=${DNSKEY_NAME}.key
-PRIKEY_NAME=${DNSKEY_NAME}.private
-ZONE_SECRET=$(cat $PWD/secrets/$PRIKEY_NAME | yj | jq -r .Key)
+# dnssec key generation was temporarily moved to 'teuthology_nameserver' ansible role due to encryption algorithm deprication on Tumbleweed
+#DNSKEY_NAME=$(make_dns_key $ZONE_DOMAIN $PWD/secrets)
+#PUBKEY_NAME=${DNSKEY_NAME}.key
+#PRIKEY_NAME=${DNSKEY_NAME}.private
+#ZONE_SECRET=$(cat $PWD/secrets/$PRIKEY_NAME | yj | jq -r .Key)
```
If it is not used anymore, just delete it

kshtsk replied:
and the function make_dns_key is not required anymore

gekios replied:
Since the algorithm will be deprecated everywhere in the future maybe we will go back to creating the key in the script so that's why I didn't remove it.

kshtsk replied:
no... there is no way back, please go ahead clean it up

kshtsk commented on deploy-teuthology:
```diff
@@ -173,19 +173,17 @@ fi
-pubkey_name: $PUBKEY_NAME
-prikey_name: $PRIKEY_NAME
 END
 scp $sshopts $PWD/secrets/${DNSKEY_NAME}* $LOGIN_USER@$target:
```
This code is senseless now, since ansible is generating the keys.

gekios replied:
yep I missed to remove this line

kshtsk commented on roles/teuthology_nameserver/tasks/dnssec-keygen.yml:
```diff
@@ -0,0 +1,44 @@
+  register:
+    dns_keygen
+  tags:
+    - dnssec_keygen
```
Since the tags are used in outside import_tasks then dnssec_keygen tag is redundant for each task of this file.

kshtsk commented on roles/teuthology_nameserver/tasks/dnssec-keygen.yml:
```diff
@@ -0,0 +1,44 @@
+    - dnssec_keygen
+
+- name: Get key secret
+  shell: "grep 'Key:' {{ dns_keygen.stdout }}.private | cut -d ' ' -f 2"
```
Since the private file is yaml file I hope ansible able to parse the yaml.

kshtsk replied:
Yeah... there is a filter `from_yaml`

gekios replied:
Isn't it more efficient what way? I mean we would need to have an extra step and var to get the key and use `from_yaml` and then save it to `secret`

kshtsk replied:
What do you call efficient? grep+cut?

gekios replied:
Done.

kshtsk commented on roles/teuthology_nameserver/tasks/dnssec-keygen.yml:
```diff
@@ -0,0 +1,30 @@
+  shell: cat {{ dns_keygen.stdout }}.private
+  register: prikey_out
+
+- name: Get prikey secret
+  set_fact: 
+    secret: "{{ (prikey_out.stdout | from_yaml).Key }}"
```
Wouldn't it be better do this in the "Set key vars" step?

kshtsk replied:
That looks great btw.

kshtsk commented on roles/teuthology_nameserver/tasks/dnssec-keygen.yml:
```diff
@@ -0,0 +1,30 @@
+
+- name: Fetch the keys to secrets local path
+  fetch:
+    src: /home/{{ ansible_ssh_user }}/{{ item }}
```
`/home/{{ ansible_ssh_user}}` prefix is extra.

kshtsk replied:
```TASK [teuthology_nameserver : Fetch the keys to secrets local path] **************************************
task path: /home/kyr/suse/teuthology-deploy/roles/teuthology_nameserver/tasks/dnssec-keygen.yml:22
failed: [51.91.148.55] (item=Ksbg.suse.de.+157+10241.key) => changed=false 
  ansible_loop_var: item
  invocation:
    module_args:
      src: /home/root/Ksbg.suse.de.+157+10241.key
  item: Ksbg.suse.de.+157+10241.key
  msg: 'file not found: /home/root/Ksbg.suse.de.+157+10241.key'
```

gekios replied:
Indeed my bad cause I was using `opensuse` user so didn't notice. I changed it to `ansible_env.HOME` and it works as it should

Commit:
ecp userdata: Enabling root user

Setting `disable_root: false` in ecp userdata, like ovh, because otherwise deployment doesn't work
since root is default user in deploy-teutology script.

Signed-off-by: Georgios Kyratsas <gkyratsas@suse.com>

gekios commented:
I tested deployment and from my pov it works as expected and the pulpito issue that I noticed is unrelated ( see https://tracker.ceph.com/issues/47631?issue_count=2&issue_position=1&next_issue_id=17592)

kshtsk commented:
@gekios I have created another PR where I add tsig-keygen support and which I was able to run successfully on OVH against opensuse-15.1 #13 
It is based on your changes, so I suggest just close this PR.
